pennylane/ops/op_math/condition.py

```diff
@@ -666,17 +681,42 @@ def _get_cond_qfunc_prim():
     cond_prim.multiple_results = True
 
     @cond_prim.def_impl
-    def _(conditions, *args_and_consts, jaxpr_branches, n_consts_per_branch, n_args):
-
-        args = args_and_consts[:n_args]
-        consts_flat = args_and_consts[n_args:]
+    def _(*all_args, jaxpr_branches, n_consts_per_branch, n_args):
+        n_branches = len(jaxpr_branches)
+        conditions = all_args[:n_branches]
+        args = all_args[n_branches : n_branches + n_args]
+        consts_flat = all_args[n_branches + n_args :]
+
+        # Find predicates that use mid-circuit measurements. We don't check the last
+        # condition as that is always `True`.
+        mcm_conditions = [pred for pred in conditions[:-1] if isinstance(pred, MeasurementValue)]
+        if len(mcm_conditions) != 0:
+            if len(mcm_conditions) != len(conditions) - 1:
+                raise ConditionalTransformError(
+                    "Cannot use qml.cond with a combination of mid-circuit measurements "
+                    "and other classical conditions as predicates."
+                )
+            conditions = _get_mcm_predicates(mcm_conditions)
 
         start = 0
         for pred, jaxpr, n_consts in zip(conditions, jaxpr_branches, n_consts_per_branch):
             consts = consts_flat[start : start + n_consts]
             start += n_consts
             if pred and jaxpr is not None:
-                return jax.core.eval_jaxpr(jaxpr.jaxpr, consts, *args)
+                if isinstance(pred, qml.measurements.MeasurementValue):
+                    with qml.queuing.AnnotatedQueue() as q:
+                        out = jax.core.eval_jaxpr(jaxpr.jaxpr, consts, *args)
+
+                    if len(out) != 0:
+                        raise ConditionalTransformError(
+                            "Only quantum functions without return values can be applied "
+                            "conditionally with mid-circuit measurement predicates."
+                        )
+                    for wrapped_op in q:
+                        Conditional(pred, wrapped_op.obj)

```

Comment:
Sorry, this might be me still getting used to the `cond` logic, but why do we need to queue these additional `wrapped_op`s for MCM predicates but not otherwise?

Reply:
That's because in pennylane's current state, the only way to know if there are mcm-controlled conditional operators is the presence of `Conditional`s in the tape. We can't inspect an MCM value mid-execution, so `qml.cond` doesn't know which branch to pick. This is the same reason why we don't support mixing MCM predicates and other classical predicates.

Reply:
Ahh, okay, this makes a lot of sense :facepalm: Thanks!